syntax = "proto3";

package Autograder;

service Autograder {
    rpc Grade(SubmissionRequest) returns (SubmissionResponse) {}
}

service LoadBalancer {
    rpc Submit(SubmissionRequest) returns (SubmissionResponse) {}
}

message SubmissionRequest {
    int32 task_id = 1;
    string source_code = 2;
}

message SubmissionResponse {
    Status status = 1;
    string output = 2;
}

/* Response status */
enum Status {
    OK = 0;
    ERROR = 1;
}
